enhancements/cnf-update-reference-config-hub-side-templ.md

```diff
@@ -0,0 +1,187 @@
+
+
+## Motivation
+
+* With large fleets of clusters, it can quickly become hard to maintain per-site configurations.
+
+* Offer users an example for how to reduce the number of policies on their hub clusters. This would make the cluster’s deployment and maintenance process a lot less cumbersome.
+
+* Customers are already making use of hub side templating, so we should cover this scenario in our reference configuration.
+
+
+### Goals
+
+* Update the existing reference configuration with the following:
+  * hub side template examples and references to its documentation
+  * PolicyGenTemplates that use templates instead of hard-coded configuration
+  * ConfigMaps that contain the configuration needed for the used PGTs
+* The user can include the desired configuration data in SiteConfig and SiteConfig Generator will create the needed ConfigMaps
+* Explain if and how the ZTP workflow is impacted by switching to using hub side templates
+
+### Non-Goals
+* hub side templating for SiteConfig
+* replacing all the content of PGT with templates
+
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-9171)
+
+## Proposal
+
+* Define configuration groups (Ex: hardware, network, ...)
+
+* Add configMap examples for the above defined groups
+
+* Include examples of hub side template with both per-cluster and per-group values being substituted
+
+* Update one of the SNO development validation repos to use the hub side templeting reference configuration
+
+* Update the siteConfig CRD to include the desired configuration under a new optional **site-data** section.
+  * with this new addition, SiteConfig will support maintaining per-site data that will be used in hub side templating.
+  * an alternative to populating the **site-data** section in the SiteConfig CR is to hold site-specific template data in the ZTP GitOps repo, as ConfigMaps (this template data will be synced to the hub cluster).
+    * using the new **site-data** section enables the user to keep all the site specific configuration in one CR rather than maintaining SiteConfig and per-site ConfigMaps in Git.
+  * if the user wants to update any site specific configuration, they can do so by modifying the corresponding data under the **site-data**.
+    * the SiteConfig Generator will update the ConfigMaps on the hub cluster with the new configuration
+
+```yaml
+apiVersion: ran.openshift.io/v1
+kind: SiteConfig
+metadata:
+  name: "sno-001"
+  namespace: "sno-001"
+spec:
+  ...
+  site-data:
+    networkNamespace: "abcdefg"
+    vlan-1: 140
+  ...
+```
+
+* Update the SiteConfig Generator to create the corresponding ConfigMaps. The generated ConfigMaps will be named as ztp-<SNO_name>-configMap and will be created under the spoke's namespace.
```

Comment:
As the resource used by the hub templating function must be in the same namespace as the policy namespace, the configmaps should be created in the site policies' namespace instead of spoke namespaces. In our reference site PGT, it's under ztp-site namespace. Unless we assume that users will create site specific policies in ztp-site, Siteconfig generator will need to know the namespace for the site policies.

Reply:
That's right, from the ACM docs on hub templating: "You can only reference namespaced Kubernetes objects that are in the same namespace as the Policy resource."
An option would be to also include the configMap's namespace in this new site-data section :thinking: